crates/rover-client/src/error.rs

```diff
@@ -217,6 +214,32 @@ fn operation_check_error_msg(check_response: &CheckResponse) -> String {
     )
 }
 
+fn downstream_check_error_msg(downstream_blocking_variants: &Vec<String>) -> String {
+    let variants = downstream_blocking_variants.join(",");
+    let plural = match downstream_blocking_variants.len() {
+        1 => "",
+        _ => "s",
+    };
+    format!(
+        "The downstream check task has encountered check failures for at least these blocking downstream variant{}: {}.",
+        plural,
+        variants,
+    )
+}
+
+fn other_check_task_failure_msg(has_build_task: &bool, has_downstream_task: &bool) -> String {
+    let succeeding_tasks = match (*has_build_task, *has_downstream_task) {
+        (false, false) => "The operations task",
+        (true, false) => "The build and operations tasks",
+        (true, true) => "The build, operations, and downstream tasks",
+        (false, true) => unreachable!("Can't have a downstream task without a build task"),
```

Comment:
i wonder if this is something we could express in the schema? perhaps downstream tasks are nested under build task results?

Reply:
We could potentially nest dependent tasks under their dependencies, although it becomes a bit trickier if dependents have multiple dependencies/the task graph gets more complicated. (It would be nice to represent the relationship in some programmatic fashion though in the API.)